pkg/integrations/v4/manager.go

```diff
@@ -332,9 +332,23 @@ func (mgr *Manager) handleFileEvent(event *fsnotify.FileEvent) {
 		WithField("file_name", event.Name)
 	elog.Debug("Received File event.")
 
-	isDelete := event.IsDelete() || event.IsRename()
-	isCreate := event.IsCreate()
-	isWrite := isCreate || event.IsModify()
+	var eDelete, eCreate, eWrite, eRename bool
+	if event.Op&fsnotify.Write == fsnotify.Write {
+		eWrite = true
+	}
+	if event.Op&fsnotify.Remove == fsnotify.Remove {
+		eDelete = true
+	}
+	if event.Op&fsnotify.Create == fsnotify.Create {
+		eCreate = true
+	}
+	if event.Op&fsnotify.Rename == fsnotify.Rename {
+		eRename = true
+	}
+
+	isDelete := eDelete || eRename
```

Comment:
Just a question, why rename == delete?

Reply:
That was the old logic in the code so was trying not to change behavior.
```
isDelete := event.IsDelete() || event.IsRename()```